### PR TITLE
Feature/display git branch

### PR DESCRIPTION
**Pull request by Zorin95670**
Display current git branch of project in the text view.

### CODE REVIEW

**Review by justinehell — CHANGES_REQUESTED** [review]

File: cypress/e2e/homepage.feature [hunk 1]
```diff
@@ -1,18 +1,19 @@
...
     Given I clear cache
     And I visit the "/"
 
     When I click on "[data-cy=\"create-empty-project\"]"
-    Then I expect current url is "/modelizer/project-[a-f0-9]{8}/model"
+    Then I expect current url is "modelizer/project-[a-f0-9]{8}/model"
     And  I extract "project-[a-f0-9]{8}" from url in field "projectName" of context
-    And I expect localstorage field "projects" is '{"{{ projectName }}":{"id":"{{ projectName }}"}}'
+
+  Scenario: Create project should redirect to project page and save it in browser cache
```
[the remark this answers is not in the document]
Comment (Zorin95670) [reply cypress/e2e/homepage.feature]: done

File: src/components/card/GitBranchCard.vue [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+let updateRemoteSubscription;
+const route = useRoute();
+const loading = ref(true);
+const currentBranch = ref('');
+
+function updateBranch() {
+  loading.value = true;
+  getCurrentBranch(route.params.projectName).then((branch) => {
+    currentBranch.value = branch;
+  }).finally(() => {
```
[the remark this answers is not in the document]
Comment (Zorin95670) [reply src/components/card/GitBranchCard.vue:34]: Yes completely :)
We will do it later =D

File: cypress/support/step_definitions/context.js [hunk 3]
```diff
@@ -4,19 +4,15 @@ Before(() => {
   cy.context = {};
 });
 
-Given('I clear cache', () => {
+Given('I clear cache', async () => {
```
Comment (justinehell) [inline cypress/support/step_definitions/context.js:7]: why use `async` without an `await` ?

**Review by justinehell — CHANGES_REQUESTED** [review]

File: cypress/support/step_definitions/context.js (hunk 3 above)
Comment (Zorin95670) [reply cypress/support/step_definitions/context.js]: Forgot to remove, thank's for the report, done :)

**Review by justinehell — DISMISSED** [review]
👍

**Review by k-y0u — CHANGES_REQUESTED** [review]

File: src/composables/events/GitEvent.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import { Subject } from 'rxjs';
+
+/**
+ * Represent a rxjs Event object to emit and to receive events about update remote on git.
```
[the remark this answers is not in the document]
Comment (Zorin95670) [reply src/composables/events/GitEvent.js:4]: done

File: tests/unit/components/card/GitBranchCard.spec.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+  });
+
+  describe('Test props initialization', () => {
+    describe('Test prop: currentBranch', () => {
+      it('Should match "main"', () => {
+        expect(wrapper.vm.currentBranch).toStrictEqual('main');
+      });
+    });
+    describe('Test prop: loading', () => {
+      it('Shouldbe false', () => {
```
[the remark this answers is not in the document]
Comment (Zorin95670) [reply tests/unit/components/card/GitBranchCard.spec.js:50]: done

File: tests/unit/components/card/GitBranchCard.spec.js [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+  });
+
+  describe('Test hook function: onMounted', () => {
+    it('should subscribe UpdateRemoteEvent', () => {
+      expect(subscribe).toHaveBeenCalledTimes(1);
+    });
+  });
+
+  describe('Test hook function: onUnmounted', () => {
+    it('should unsubscribe UpdateRemoteEvent', () => {
```
[the remark this answers is not in the document]
Comment (Zorin95670) [reply tests/unit/components/card/GitBranchCard.spec.js:63]: done

**Review by k-y0u — APPROVED** [review]

**Review by justinehell — APPROVED** [review]

**Pull request merged**